src/lib/event_mapper/resource_type_resolver.py

```diff
@@ -0,0 +1,32 @@
+from ..core.constants import SettingConfigResourceTypes as types
+
+
+class ResourceTypeResolver:
+    _source_detail_type_map = {
```

Comment:
I personally don't like that resolver structure and resolution algorithm, but please advise if that can be done more beautifully in Python without if-else structures 🙃